src/mgos_mqtt_conn.c

```diff
@@ -638,7 +638,10 @@ bool mgos_mqtt_conn_unsub(struct mgos_mqtt_conn *c, const char *topic) {
     if (0 == strcmp(s->topic.p, topic)) {
```

Comment:
you cannot strcmp mg_str.p, it may not be nul-terminated. mg_str is a StringPiece, essentially (you may be familiar with that).

Reply:
Done. Code still runs clean on esp8266 and ubuntu platform.